tests/dist.rs

```diff
@@ -496,24 +498,50 @@ fn uninstall(
     Ok(())
 }
 
+#[derive(Copy, Clone, Debug)]
+enum Compressions {
+    GZOnly,
+    AddXZ,
+    AddZStd,
+    AddBoth,
+}
+use Compressions::*;
+
+impl Compressions {
+    fn enable_xz(self) -> bool {
+        matches!(self, AddXZ | AddBoth)
+    }
+
+    fn enable_zst(self) -> bool {
+        matches!(self, AddZStd | AddBoth)
+    }
+}
+
 fn setup(
     edit: Option<&dyn Fn(&str, &mut MockChannel)>,
-    enable_xz: bool,
+    comps: Compressions,
     f: &dyn Fn(&Url, &ToolchainDesc, &InstallPrefix, &DownloadCfg<'_>, &temp::Cfg),
 ) {
     let dist_tempdir = tempfile::Builder::new().prefix("rustup").tempdir().unwrap();
     let mock_dist_server = create_mock_dist_server(dist_tempdir.path(), edit);
     let url = Url::parse(&format!("file://{}", dist_tempdir.path().to_string_lossy())).unwrap();
-    setup_from_dist_server(mock_dist_server, &url, enable_xz, f);
+    setup_from_dist_server(
+        mock_dist_server,
+        &url,
+        comps.enable_xz(),
+        comps.enable_zst(),
+        f,
+    );
 }
 
 fn setup_from_dist_server(
     server: MockDistServer,
     url: &Url,
     enable_xz: bool,
```

Comment:
You've done this lovely enum above, and then not here.

Reply:
That'd be because I'm a plonker.  Moved the logic in one level.  I'm not going to push it into `server.write()` for now though.  I think that refactor can wait for the vec/otherwise refactor later.